instant-xml-macros/src/lib.rs

```diff
@@ -50,66 +52,46 @@ pub(crate) fn get_namespaces(
     (default_namespace, other_namespaces)
 }
 
-pub(crate) fn retrieve_field_attribute(name: &str, input: &syn::Field) -> Option<FieldAttribute> {
-    if let Some(list) = retrieve_attr_list(name, &input.attrs) {
-        match list.nested.first() {
-            Some(NestedMeta::Lit(Lit::Str(v))) => {
-                return Some(FieldAttribute::Namespace(v.value()));
-            }
+pub(crate) fn retrieve_field_attribute(input: &syn::Field) -> Option<FieldAttribute> {
+    match retrieve_attr_list(&input.attrs) {
+        Some((Some(list), name)) if name.as_str() == "namespace" => match list.nested.first() {
+            Some(NestedMeta::Lit(Lit::Str(v))) => Some(FieldAttribute::Namespace(v.value())),
             Some(NestedMeta::Meta(Meta::Path(v))) => {
                 if let Some(ident) = v.get_ident() {
-                    return Some(FieldAttribute::PrefixIdentifier(ident.to_string()));
+                    Some(FieldAttribute::PrefixIdentifier(ident.to_string()))
+                } else {
+                    panic!("unexpected parameter");
                 }
             }
-            _ => (),
-        };
+            _ => panic!("unexpected parameter"),
+        },
+        Some((None, name)) if name.as_str() == "attribute" => Some(FieldAttribute::Attribute),
+        None => None,
+        _ => panic!("unexpected parameter"),
     }
-    None
 }
 
-pub(crate) fn retrieve_attr(name: &str, attributes: &Vec<syn::Attribute>) -> Option<bool> {
+fn retrieve_attr_list(attributes: &Vec<syn::Attribute>) -> Option<(Option<syn::MetaList>, String)> {
```

Comment:
I don't think the values returned here needed to be owned?

Reply:
`parse_meta` is returning the value, so we can not return the reference for those values here.

Reply:
Ah, fair.